### DISCUSSION TITLE
Support `RENOVATE_LOG_FILE` in addition to `LOG_FILE` env var

### DISCUSSION THREAD

**Discussion by simonhaenisch** [renovatebot/renovate#30635] [Suggest an Idea]
### Tell us more.

While upgrading to v38, I saw the change regarding removal of the `logFile` option. I thought we were good until I noticed we were actually setting `RENOVATE_LOG_FILE` instead of `LOG_FILE` (same for `LOG_FILE_LEVEL` and `LOG_CONTEXT`). I realized that `RENOVATE_LOG_FILE` was actually configuring the `logFile` config option via env, thus not being equivalent to `LOG_FILE`.

Would it make sense to still support `RENOVATE_LOG_FILE` etc. additionally? Asking because just `LOG_FILE` seems a bit generic and could be used by other tools as well, which someone had already reported in #30453.

**Comment by rarkins** [comment]
Yes I think we could support a RENOVATE_ prefix for all current LOG_ prefixes including LOG_LEVEL. You're welcome to submit a PR if you like

**Reply by 1oglop1** [reply]
@rarkins It is quite difficult to find this discussion. 
I found it by accident while submitting a bug report, because I see this as a regression.
IMO all env variables should have `RENOVATE_` prefix to avoid conflicts in places where more programs run. 

But since this will likely be another breaking change it is probably suitable for [v39 planning](https://github.com/renovatebot/renovate/issues/25608) 


By quickly searching the code for `process.env` it seems like a mix of approaches for env vars.
Are there any guidelines for implementing environment variables?
Eg. variables affecting renovate binary have prefix `RENOVATE_` but variables affecting 3rd party hosts eg. GitHub does not have the prefix `GITHUB_TOKEN` not `RENOVATE_GITHUB_TOKEN`.

https://github.com/search?q=repo%3Arenovatebot%2Frenovate+process.env&type=code

**Reply by rarkins** [reply]
Almost all env vars in Renovate should have `RENOVATE_` prefixes, unless they are variable names from other systems which we cannot change. The `LOG_` ones are an exception right now but we can also support `RENOVATE_LOG_` too without it being breaking for those using only `LOG_`.

**Reply by 1oglop1** [reply]
Thanks for the swift response!
This is great news! 
I'd like to fix this myself because I plan a few more tiny contributions. Could you point me to the relevant place in the code or do you expect to use the null check for `process.env.LOG_LEVEL ?? process.env.RENOVATE_LOG_LEVEL` somewhere around here?

https://github.com/renovatebot/renovate/blob/2dbfe4d872e3e5db970386a737568a102c310285/lib/logger/index.ts#L21

**Reply by rarkins** [reply]
Yes. And in the next major release we should reverse that order to make the renovate prefix preferred
